### PR TITLE
implement request/response anyOf checkers

### PR DESCRIPTION
**Pull request by ciprian-tibulca**

### CODE REVIEW

File: checker/check-request-property-any-off-updated.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
+package checker
+
+import (
+	"fmt"
+
+	"github.com/tufin/oasdiff/diff"
+)
+
+func RequestPropertyAnyOffUpdated(diffReport *diff.Diff, operationsSources *diff.OperationsSourcesMap, config Config) Changes {
```
Comment (gssbzn) [inline checker/check-request-property-any-off-updated.go]:
```suggestion
func RequestPropertyAnyOfUpdated(diffReport *diff.Diff, operationsSources *diff.OperationsSourcesMap, config Config) Changes {
```

Comment (ciprian-tibulca) [reply checker/check-request-property-any-off-updated.go]: good catch, thank you. 
Fixed this typo (in few places)

File: checker/check-request-property-any-off-updated_test.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+	s2, err := open("../data/checker/request_property_any_of_added_revision.yaml")
+	require.NoError(t, err)
+
+	d, osm, err := diff.GetWithOperationsSourcesMap(getConfig(), s1, s2)
+	require.NoError(t, err)
+	errs := checker.CheckBackwardCompatibilityUntilLevel(singleCheckConfig(checker.RequestPropertyAnyOfUpdated), d, osm, checker.INFO)
+
+	require.Len(t, errs, 2)
+
+	require.Equal(t, checker.ApiChange{
```
Comment (gssbzn) [inline checker/check-request-property-any-off-updated_test.go]: consider using `ElementsMatch` it'll check two arrays are the same without caring about order of elements

File: checker/localizations_src/en/messages.yaml [hunk 3]
```diff
@@ -166,3 +166,11 @@ response-property-pattern-added: the %s response's property pattern %s was added
 response-property-pattern-removed: the %s response's property pattern %s was removed for the status %s
 response-property-default-value-changed: the %s response's property default value changed from %s to %s for the status %s
 response-body-default-value-changed: the response body %s default value changed from %s to %s for the status %s
+request-body-any-of-added: added '%s' to the request body 'anyOf' list
```
Comment (gssbzn) [inline checker/localizations_src/en/messages.yaml]: not sure what the standard practice is on the copy but `%q` adds quotes 
```suggestion
request-body-any-of-added: added %q to the request body 'anyOf' list
```

File: checker/check-request-property-any-off-updated.go [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+						Operation:   operation,
+						OperationId: operationItem.Revision.OperationID,
+						Path:        path,
+						Source:      source,
+					})
+				}
+
+				CheckModifiedPropertiesDiff(
+					mediaTypeDiff.SchemaDiff,
+					func(propertyPath string, propertyName string, propertyDiff *diff.SchemaDiff, parent *diff.SchemaDiff) {
```
Comment (gssbzn) [inline checker/check-request-property-any-off-updated.go]: does this function requires the closure? to simplify the method length could it be moved outside? same on the other similar anonymous functions

File: checker/check-request-property-any-of-updated.go [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+						OperationId: operationItem.Revision.OperationID,
+						Path:        path,
+						Source:      source,
+					})
+				}
+
+				if mediaTypeDiff.SchemaDiff.AnyOfDiff != nil && mediaTypeDiff.SchemaDiff.AnyOfDiff.Deleted > 0 {
+					result = append(result, ApiChange{
+						Id:    "request-body-any-of-removed",
+						Level: ERR,
```
Comment (ciprian-tibulca) [inline checker/check-request-property-any-of-updated.go:51]: breaking change

File: checker/check-request-property-any-of-updated.go [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+								Path:        path,
+								Source:      source,
+							})
+						}
+
+						if propertyDiff.AnyOfDiff.Deleted > 0 {
+							result = append(result, ApiChange{
+								Id:    "request-property-any-of-removed",
+								Level: ERR,
+								Text: fmt.Sprintf(
```
Comment (ciprian-tibulca) [inline checker/check-request-property-any-of-updated.go:88]: breaking change

File: checker/localizations_src/en/messages.yaml (hunk 3 above)
Comment (ciprian-tibulca) [reply checker/localizations_src/en/messages.yaml]: %q will add double-quotes (`a double-quoted string safely escaped with Go syntax`) which won't be consistent with the rest of the messages oasdiff already has

File: checker/check-request-property-any-off-updated_test.go (hunk 2 above)
Comment (ciprian-tibulca) [reply checker/check-request-property-any-off-updated_test.go]: done, thanks for the suggestion

File: checker/check-request-property-any-off-updated.go (hunk 4 above)
Comment (ciprian-tibulca) [reply checker/check-request-property-any-off-updated.go]: it's a pattern used across all checkers, and this func is accessing quite a few vars from outside, so the code will look fairly the same in terms of complexity if extracted

File: checker/localizations_src/en/messages.yaml (hunk 3 above)
Comment (blva) [reply checker/localizations_src/en/messages.yaml]: on this, one thing you can do is to keep %s and print with `ColorizedValue` which is something I've seen across the repo

Comment (ciprian-tibulca) [reply checker/localizations_src/en/messages.yaml]: good idea, done

File: checker/checker_breaking_test.go [hunk 7]
```diff
@@ -630,6 +630,28 @@ func TestBreaking_RequestBodyMediaTypeRemoved(t *testing.T) {
 	require.Equal(t, "removed the media type application/json from the request body", errs[0].GetText())
 }
 
+// BC: removing 'anyOf' schema from the request body or request body property
```
Comment (reuvenharrison) [inline checker/checker_breaking_test.go]: add suffix: "is breaking"

File: checker/checker_breaking_test.go [hunk 8]
```diff
@@ -630,6 +630,28 @@ func TestBreaking_RequestBodyMediaTypeRemoved(t *testing.T) {
...
+	require.Equal(t, "request-body-any-of-removed", errs[0].GetId())
+	require.Equal(t, checker.ERR, errs[0].GetLevel())
+	require.Equal(t, "removed 'Rabbit' from the request body 'anyOf' list", errs[0].GetText())
+
+	require.Equal(t, "request-property-any-of-removed", errs[1].GetId())
+	require.Equal(t, checker.ERR, errs[1].GetLevel())
+	require.Equal(t, "removed 'Breed3' from the '/anyOf[#/components/schemas/Dog]/breed' request property 'anyOf' list", errs[1].GetText())
+}
+
 // BC: removing 'oneOf' schema from the request body or request body property
```
Comment (reuvenharrison) [inline checker/checker_breaking_test.go]: add suffix: "is breaking"

File: checker/checker_breaking_test.go (hunk 7 above)
Comment (ciprian-tibulca) [reply checker/checker_breaking_test.go]: done, thank you

File: checker/checker_breaking_test.go (hunk 8 above)
Comment (ciprian-tibulca) [reply checker/checker_breaking_test.go]: done, thank you

**Review by reuvenharrison — APPROVED** [review]

**Pull request merged**